Fixes #981 - adds images to explain label placement option

SrNetoChan opened the pull request:
I also did some reformatting,but the important part is in Labels properties. I will squash all commits during merge. @DelazJ and @yjacolin Reviews as appreciated, thanks.

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -90,22 +90,21 @@ you use the current map canvas scale as boundary of the range visibility.
-query.
+Under the **Provider Feature Filter** frame, the :index:`Query Builder` allows
+you to define a subset of the features in the layer using a SQL-like WHERE
+clause and to display the result in the main window.  As long as the query is
```
double space before `As long as`. (not the only place there are double space)

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -985,62 +984,70 @@ The following steps describe simple labeling without using the
-For line vector layers you can include line directions symbols. There are options
-specifying the type of symbol and the symbol placement.
+In the :guilabel:`Text`tab, you can define the :guilabel:`Font`,
+:guilabel:`Style`, and :guilabel:`Size` of your labels' text (see
```
Is a comma needed before `and`? (nota: i'm following french grammar rules. I don't recall how it's in english)

SrNetoChan replied:
This is called the oxford comma. It states that one should use an comma before and in the final part of a list.

https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/

DelazJ replied:
Ah ok! Thanks for teaching me that. So both may be correct. I'll bookmark your link: might be helpful for other rules.

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -985,62 +984,70 @@ The following steps describe simple labeling without using the
-specifying the type of symbol and the symbol placement.
+In the :guilabel:`Text`tab, you can define the :guilabel:`Font`,
+:guilabel:`Style`, and :guilabel:`Size` of your labels' text (see
+Figure_labels_1_). There are options available to set the labels'
```
could we begin removal of this kind of numbered references?

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -985,62 +984,70 @@ The following steps describe simple labeling without using the
+the possibility to render the text as 'All uppercase', 'All lowercase'
+or 'Capitalize first letter'. In :guilabel:`Spacing`, you can change the space
+between words and between individual letters. Finally, use the
+:guilabel:`Blend mode`option to determine how you labels will mix with the
```
* require a space before `option`, otherwise it won't be rendered in blue
* you**r** labels`

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -985,62 +984,70 @@ The following steps describe simple labeling without using the
+or 'Capitalize first letter'. In :guilabel:`Spacing`, you can change the space
+between words and between individual letters. Finally, use the
+:guilabel:`Blend mode`option to determine how you labels will mix with the
+map features bellow them (see more about it in :ref:`blend-modes`).
```
below

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -985,62 +984,70 @@ The following steps describe simple labeling without using the
+In the :guilabel:`Formatting` tab, you can define a character for a line break
+in the labels with the :guilabel:`Wrap on character` option. You can also
+format the :guilabel:`Line Height` and the :guilabel:`alignment`. For the
+latter, typical values are available, plus *Follow label placement*. When set
```
Not sure to really understand this sentence but note that `Follow label placement` is afaics available only for point layer so I think we should clearly mention it beforehand.

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -985,62 +984,70 @@ The following steps describe simple labeling without using the
+placed to the right of the point then the label will be left aligned.
+
+For line vector layers you can include :guilabel:`Line directions symbols`
+to help determine the lines directions. They work particularaly well when
```
particularly

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -985,62 +984,70 @@ The following steps describe simple labeling without using the
-aspect of the buffer's stroke intersects with the text's fill) and also allows
-you to make outlined text.
+To create a buffer around the labels, activate the |checkbox|
+:guilabel:`Draw text buffer` checkbox in the :guilabel:`Buffer`tab. You can
```
space needed before `tab`

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -985,62 +984,70 @@ The following steps describe simple labeling without using the
+:guilabel:`Transparency`. If the |checkbox| :guilabel:`color buffer's fill`
+checkbox is activated, it will interact with partially transparent text and
+give mixed color transparency results. Turning off the buffer fill fixes
+that issue (except where the interior aspect of the buffer's stroke
```
`fixes that issue`: is this considered as a bug in the application? I wonder if the way this sentence is written does not let think that the behaviour of `color buffer's fill` is not the one expected. Is it?

Actually, testing the `color buffer's fill` option (check/uncheck), I wonder if it changes something in the map canvas. Can some confirm, please?

SrNetoChan replied:
@DelazJ This sentence is not mine, I only did some changes. Have been doing some tests, and you can only see the difference if you make your label transparent and use a relative big font. I will try to clarify the sentence.

![screenshot from 2016-09-04 09-28-34](https://cloud.githubusercontent.com/assets/3607161/18229895/0497edda-7282-11e6-8a1a-ec44eed8bc44.png)

DelazJ replied:
Yes, no worries. I know it's not yours. And thanks for clarifying it.

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -1123,88 +1130,95 @@ feature using a data defined list of prioritised positions.
-able to set the X and Y offset distances between the points and their labels
-and can alter the angle of the label placement with the :guilabel:`Rotation` setting.
-Thus, placement in a selected quadrant with a defined rotation is possible.
+In the |radioButtonOn| :guilabel:`Offset from point`placement, labels are
```
space missed before `placement`

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -1123,88 +1130,95 @@ feature using a data defined list of prioritised positions.
+:guilabel:`Above line`, |checkbox| :guilabel:`On line` and |checkbox|
+:guilabel:`Below line`. It's possible to select several options at once. In
+that case, QGIS will look for the optimal label position. For Parallel and
+curved placement you can also use the line orientation for the position of th
```
placement**s**?
the position of **the**

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -1123,88 +1130,95 @@ feature using a data defined list of prioritised positions.
+for the centroid.
+
+With the |radioButtonOff| :guilabel:`Using perimeter` option, the label
+will be draw next to the polygon boundary. The label will behave like the
```
drawn

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -1123,88 +1130,95 @@ feature using a data defined list of prioritised positions.
-label with the higher priority will be displayed and the others will be left out.
+In the :guilabel:`priority` section you can define the priority with which
+labels are rendered for all three vector layer types (point, line, polygon).
+This placement option interact with labels of the other vector layers in the
```
interacts

DelazJ commented:
Some general comments:
* while breaking lines to 80 characters, you inserted sometimes a second space in the text. I'm afraid Transifex does not keep the translation (given that the text does no longer match at 100%)
* could you use this PR to get rid of things like figure_labels_1 in the part you are fixing, as discussed at Bonn?
Thanks

SrNetoChan commented:
I'm not sure we should start "randomly" removing figures numbers without a solution on how to reference them in the text. After a decision, I would prefer to do it for all files in one single PR.

SrNetoChan commented:
I will fix the rest of the issues tomorrow.

SrNetoChan commented:
@DelazJ I have fixed all the issues you have reported (except for the figures numbers ones), fixed some grammar in the label section, and improved the way label expressions examples look. Could you please give a look, and tell me what you think?

Thanks!

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -1153,24 +1156,25 @@ For the |radioButtonOn| :guilabel:`Parallel` and |radioButtonOff|
-label. Additionally, you can define a :guilabel:`Maximum angle between
-curved characters` when selecting the |radioButtonOff| :guilabel:`Curved`
-option (see Figure_labels_2_).
+curved placements options, you can also use the line orientation for the
```
not sure placement should be in plural. Otherwise, see lines 1027 and 1175

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -1153,24 +1156,25 @@ For the |radioButtonOn| :guilabel:`Parallel` and |radioButtonOff|
 
-For all three placement options you can set up a minimum distance for repeating
-labels. The distance can be in ``mm`` or in ``map units``.
+For all three placement options, in :guilabel`Repeat`, you can set up a
```
:guilabel:

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -1319,52 +1325,85 @@ some descriptive text and the function ``$area`` in combination with
    "name" || ', ' || "place"
 
-   -> John Smith, Paris
+   Returns:
```
returns:: (with double colon)

SrNetoChan replied:
@DelazJ, since this is a result, and not code, I wanted to render it as a quote only. Do you think I should use :: anyway?

DelazJ replied:
What do you mean by _I wanted to render it as a quote only_? A simple text as in the commit (also be cautious about the indentation - the current commit may not render as you expect). 
A risk is that people translate it (some translators do not see where the text is really displayed in the doc before translating).
Maybe using double ``?

DelazJ replied:
I think that having a frame around the code and then the result (when they are not together as for http://docs.qgis.org/testing/en/docs/user_manual/introduction/getting_started.html#command-line-options) helps reader to easily visualize the two things.
For example, in http://docs.qgis.org/testing/en/docs/documentation_guidelines/writing.html#label-reference, I don't find the return really visible at the first reading.

DelazJ commented:
Made a quick review. few comments, otherwise LGTM

SrNetoChan commented:
You have a very good eye!!

SrNetoChan commented:
I will change it as you suggested, using :: . Note that I needed to use code-block in the code because concat() was not rendering correctly in some of the examples.
